python/res/job_queue/workflow_runner.py

```diff
@@ -20,12 +20,20 @@ def __init__(self, workflow, ert=None, context=None):
 
         self.__context = context
         self.__workflow_result = None
+        self.__workflow_executor = ThreadPoolExecutor(max_workers=1)
+        self.__workflow_job = None
+        self.__workflow_exception = None
+
+    def __enter__(self):
+        self.run()
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.wait()
 
     def run(self):
-        workflow_thread = Thread(name="ert_gui_workflow_thread")
-        workflow_thread.setDaemon(True)
-        workflow_thread.run = self.__runWorkflow
-        workflow_thread.start()
+        #should we check if it's already running?
```

Comment:
If that is intended usage then please go ahead ;)

Reply:
ok

Reply:
Comment removed!